docs/frontend-system/building-plugins/02-testing.md

```diff
@@ -93,98 +95,57 @@ A number of features (frontend extensions and overrides) are also accepted by th
 import { createExtensionTester } from '@backstage/frontend-test-utils';
-import { indexPageExtension, detailsPageExtension } from './plugin';
+import { indexPageExtension, indexPageHeader } from './plugin';
 
 describe('Index page', async () => {
-  it('should link to the details page', () => {
-    createExtensionTester(indexPageExtension)
-      // Adding more extensions to the preset being tested
-      .add(detailsPageExtension)
-      .render();
+  it('should link to the index page with header', async () => {
+    const tester = createExtensionTester(indexPageExtension)
+      // Adding the header to be rendered on the index page
+      .add(indexPageHeader);
 
-    await expect(screen.findByText('Index Page')).toBeInTheDocument();
+    await renderInTestApp(tester.reactElement());
 
-    await userEvent.click(screen.getByRole('link', { name: 'See details' }));
+    await expect(screen.findByText('Index page')).toBeInTheDocument();
+    await expect(screen.findByText('Index page header')).toBeInTheDocument();
 
-    await expect(
-      screen.findByText('Details Page'),
-    ).resolves.toBeInTheDocument();
+    expect(
+      tester.query(indexPageHeader).get(headerDataRef),
+    ).toMatchObject(/* ... */);
   });
 });
 ```
 
-### Mocking apis
-
-If your extensions requires implementation of APIs that aren't wired up by default, you'll have to add overrides to the preset of features being tested:
-
-```tsx
-import { screen } from '@testing-library/react';
-import userEvent from '@testing-library/user-event';
-import { createApiFactory } from '@backestage/core-plugin-api';
-import {
-  createExtensionOverrides,
-  configApiRef,
-  analyticsApiRef,
-} from '@backstage/frontend-plugin-api';
-import {
-  createExtensionTester,
-  MockConfigApi,
-  MockAnalyticsApi,
-} from '@backstage/frontend-test-utils';
-import { indexPageExtension } from './plugin';
-
-describe('Index page', () => {
-  it('should capture click events in analytics', async () => {
-    // Mocking the analytics api implementation
-    const analyticsApiMock = new MockAnalyticsApi();
-
-    const analyticsApiOverride = createApiExtension({
-      factory: createApiFactory({
-        api: analyticsApiRef,
-        factory: () => analyticsApiMock,
-      }),
-    });
-
-    createExtensionTester(indexPageExtension)
-      // Overriding the analytics api extension
-      .add(analyticsApiOverride)
-      .render();
-
-    await userEvent.click(
-      await screen.findByRole('link', { name: 'See details' }),
-    );
-
-    expect(analyticsApiMock.getEvents()[0]).toMatchObject({
-      action: 'click',
-      subject: 'See details',
-    });
-  });
-});
-```
+When testing multiple extensions you may sometimes want to access the output of other extensions than the main test subject. You can use the `.query(ext)` method to query a different extension that has been added to the tester, either via its ID or passing the extension directly.
```

Comment:
```suggestion
When testing multiple extensions you may sometimes want to access the output of other extensions than the main test subject. You can use the `.query(ext)` method to query a different extension that has been added to the tester, by passing the extension used with the `createExtensionTester(...).add(ext)`
```

Reply:
We dropped support for the string version to make the types nicer, happy to revisit if you think it's worth querying with just an ID, but wonder if it's worth it?

Reply:
ah yep no that's good 👍